module/grid/src/Factory/DbalDataSetFactory.php

```diff
@@ -33,6 +33,7 @@ public function create(QueryBuilder $queryBuilder): DataSetInterface
         $result->select('*');
         $result->from(sprintf('(%s)', $queryBuilder->getSQL()), 't');
         $result->setParameters($queryBuilder->getParameters(), $queryBuilder->getParameterTypes());
+        $result->distinct();
```

Comment:
Why distinct in main factory ?
IMHO use distinct in this case can by only hiding a problem not a resolving it. 
Are we sure that fetching duplicated data is not inserting  or query problem ??

Reply:
I converted it to a draft for now. We surely know it does not resolve this particular problem